### DISCUSSION TITLE
Convert Reaper .rpp file to YAML

### DISCUSSION THREAD

**Discussion by ginjo** [jamulussoftware/jamulus#2567] [General]
I'm posting this with the intention of helping others like me who searched exhaustively for a way to programmatically access data related to Jamulus recordings. I'm not sure of the best place to post this, so apologies if I picked the wrong one. I'm happy to move/repost or whatever the moderators prefer.

I've been doing a lot of automation scripting of Jamulus server on Linux. One thing that's been missing for me is a manifest of each recording session in a well-supported format. The Reaper .rpp file has all the data, and it's structured in a sane way, but there's no easy way to programmatically access the data (at least none that I've found). Maybe someday Jamulus will export this data as xml, yaml, or json?

Meanwhile, I created a shell script to convert the .rpp data to yaml. From yaml, one can easily convert to json, xml, whatever, using your favorite libraries.

https://gist.github.com/ginjo/de9cf9397b28908c68a8947be219f237

**Comment by ann0see** [comment]
Interesting!

> Maybe someday Jamulus will export this data as xml, yaml, or json

Kind of ;-). @dtinth @hoffie and a few others worked on a JSON-RPC API. You can test it with the nightly or next release I mentioned in your other issue

**Reply by hoffie** [reply]
I don't think we expose something equivalent to the .rpp files as json though.

I'm not a Reaper user myself (and well, not a user of Jamulus recordings most of the time either ;)). I guess there is a very relevant user base with Reaper, but I always found it strange to support exactly that.

**Comment by ginjo** [comment]
That also sounds interesting! I found the json-rpc interface on the latest nightly you linked. Is there some documentation for that API?

**Reply by ann0see** [reply]
Yes. See the docs/ folder in the code tab on GitHub or just download the source and read the file in docs/JSON-RPC.md

**Reply by ginjo** [reply]
Ah yes, the docs folder, thanks.   😊

**Comment by ginjo** [comment]
Been awhile, but I've been revisiting the issue of "unnamed" audio files from Jamulus recordings. I just wanted to point out to anyone who stumbles across this, that the script I linked above can be used to convert audio file names after the recording is finished. Just convert the .rpp to yaml, pass the yaml to your favorite language, iterate through the tracks and rename your files.

Also wanted to say thanks for the json-rpc interface! That will be very helpful for a project I'm currently working on.
